tests/framework/artifacts.py

```diff
@@ -463,7 +462,7 @@ def kernels(self, keyword=None):
             keyword=keyword,
         )
 
-        supported_kernels = {f"vmlinux-{sup}.bin" for sup in SUPPORTED_KERNELS}
+        supported_kernels = {f"vmlinux-{sup}.bin" for sup in select_supported_kernels()}
```

Comment:
This seems to be the only place where you would need the select_supported_kernels() logic so do you think it would be good to move the logic here instead of defining the function in utils.py? or do you see any other use case where you would need select_supported_kernels()?

Reply:
When we drop 4.14 do you plan to keep the artifact.py code as is or will it switch to calling SUPPORTED_KERNELS again?
1. If you plan to keep the select_supported_kernels() and 4.14 host support is dropped then we won't need the 5.10_no_sve and select_supported_kernels() will be simply calling SUPPORTED_KERNELS, right? and in this case select_supported_kernels() becomes a redundant function.
2. if you plan to modify artifacts.py instead to switch from select_supported_kernels() to SUPPORTED_KERNELS when we drop 4.14 host then I think having the function locally seems like a good idea.

There is some refactoring done by moving get_instance_type to utils and this is just to define select_supported_kernels() in utils.py, and its not going to be used by any other code and also going to be redundant when we drop 4.14 so, I am trying to understand the pros/cons of defining this function in utils vs keeping it here locally.

Reply:
by moving select_supported_kernels() to artifacts.py you can avoid moving get_instance_type() from utils_cpuid.py to utils.py and also avoid all the other changes to the test files which are not directly related to this PR.

Reply:
Ok, that makes a lot of sense. I'll do that.